packages/micromark-core-commonmark/dev/lib/label-end.js

```diff
@@ -39,22 +39,33 @@ const referenceCollapsedConstruct = {tokenize: tokenizeReferenceCollapsed}
 /** @type {Resolver} */
 function resolveAllLabelEnd(events) {
   let index = -1
-
+  /** @type {Array<Event>} */
+  const newEvents = []
   while (++index < events.length) {
     const token = events[index][1]
+    if (newEvents.length > 0) {
+      push(newEvents, [events[index]])
+    } else {
+      newEvents.push(events[index])
+    }
```

Comment:
This looks rather weird? Why not:

```suggestion
    push(newEvents, [events[index]])
```

Or:


```suggestion
    newEvents.push(events[index])
```

Reply:
Yeah, because push function will not push events when the array is empty, I don't know the correct way to use it. So I wrote down the strange code. Do you have any advise?

https://github.com/micromark/micromark/blob/main/packages/micromark-util-chunked/dev/index.js#L82-L89

![CleanShot 2024-11-11 at 21 56 22@2x](https://github.com/user-attachments/assets/6dec23c7-80e8-45fb-8154-f13fbe683f0e)

Reply:
I think the `newEvents.push(events[index])` is ok，do you recommend use this way if the push function need non-empty array.

Reply:
How it works is explained in the comment.

You now wrote code that always creates arrays with one item for each item. That is potentially millions of arrays. That is not a good idea.

This specific code is for other cases, where it can be fast.

If your use case is to always only ever push one item — this is not the function for you. You can use `Array#push`.

The solution *with* the specific custom push is `newEvents = push(newEvents, [events[index]])`. But, this custom push is not good for this use case. You should be able to use the regular push: `newEvents.push(events[index])`.

Reply:
OK, I will try another implementation.